app/knight.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.knight_property = knight
+        self.name = knight["name"]
+        self.power = 0
+        self.hp = 0
+        self.protection = 0
+
+    def is_item(self, key: str) -> bool:
+        if self.knight_property[key] is not None:
+            return True
+        return False
+
+    def set_power(self) -> int:

```

Comment:
Code Style Issue: The method name `set_power` is misleading because it does not actually set the power attribute of the knight. It simply returns a calculated value. Consider renaming it to something like `calculate_power`.